### PR TITLE
Fixed the display of schema information

### PR DESCRIPTION
**Pull request by srkukarni**
### Motivation
pulsar-admin schemas get currently prints out a base64 encoded gibbersh information about the schema details. This is because of the conversion of the correct schema information received from the server into bytes structure inside SchemaInfo. We change that in this pr so that the cli tool prints sane information.

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by jerrypeng** [comment]
This PR is fine but I have a larger concern about how we a treating SchemaInfo.data.  In the API/SPI its a byte[] but in other places like GetSchemaResponse, we are converting it to a String.  This seems inappropriate and can cause confusion and bugs.  Currently, I think all SchemaInfo.data for schemas that use this is actually a String stored as bytes which shouldn't break things but I still think its bad practice assume the byte[] is convertible to a String.  Either we change the SchemaInfo.data to a String or we stop making assumptions in other places that we can converted to a String. Perhaps @sijie  @codelipenghui can also comment on this

**Comment by sijie** [comment]
I am not in favor of using `String` for storing schema data. because String requires correct charsets to encode and decode. So I would suggest keeping `byte[]` for storing schema data.

GetSchemaResponse made a bad assumption on converting `byte[]` to `string`. Ideally it should be encoded using base64 because json doesn't support `byte[]` well. It is a legal BC issue that we have to face. 

that said, I don't think it is a good idea to use `GetSchemaResponse` directly. We should still convert the `GetSchemaResponse` to SchemaInfo. We should also consider fixing`GetSchemaResponse` to carry `byte[]` in base64 encoded format.

We can have a util class on displaying `SchemaInfo` in a correct format.

- For struct schema (avro, json, protobuf), we are using avro schema definition on defining the schema structure. So we can convert the `byte[]` back to the json string and display the json string in a pretty format.

- For non-struct schema, we should just use base64 for displaying schema data. 

In summary, my opinion is:

- it is a display issue. We should just fix the display issue rather than changing the return result.
- GetSchemaResponse is buggy. We should consider improving or fixing it.

**Comment by sijie** [comment]
@srkukarni - FYI I added a method in `SchemaInfo` for displaying schema definition in #4325

**Comment by srkukarni** [comment]
@sijie @jerrypeng 
Based on your feedback I have changed the pr. Please take a look again.

**Comment by jerrypeng** [comment]
This is BC breaking change but GetSchemaResource was incorrect to begin with and can cause bugs.  I am ok with this, but we should document the change as BC breaking in the release nots

**Comment by srkukarni** [comment]
@sijie sounds good. Will wait for that to merge and then use that method here

**Comment by srkukarni** [comment]
#4325

File: pulsar-common/src/main/java/org/apache/pulsar/common/schema/GetSchemaResponse.java [hunk 1]
```diff
@@ -32,6 +32,6 @@
     private long version;
     private SchemaType type;
     private long timestamp;
-    private String data;
+    private byte[] data;
```
Comment (sijie) [inline pulsar-common/src/main/java/org/apache/pulsar/common/schema/GetSchemaResponse.java]: This has to be in a new version of rest endpoint. Otherwise it is a BC problem.

**Comment by sijie** [comment]
I think there are 2 issues to addressed here.

a) fix the display issue of schema info in the admin cli.
b) fix the transport issue of schema data in rest api.

a) is using (waiting) the util function I added in #4325 

for b), I think we need to add that to new version of rest endpoints.

**Comment by srkukarni** [comment]
@sijie Have incorporated the display function.

File: pulsar-common/src/main/java/org/apache/pulsar/common/schema/GetSchemaResponse.java (hunk 1 above)
Comment (srkukarni) [reply pulsar-common/src/main/java/org/apache/pulsar/common/schema/GetSchemaResponse.java]: I have reduced the scope of this pr to just a). the b) part should be in another pr

**Comment by sijie** [comment]
run integration tests

**Comment by srkukarni** [comment]
run integration tests

**Comment by srkukarni** [comment]
run cpp tests

**Comment by srkukarni** [comment]
run cpp tests
run integration tests
run java8 tests

**Comment by srkukarni** [comment]
run integration tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run integration tests

**Comment by srkukarni** [comment]
run java8 tests
